packages/client/katello-pull-provider-migrate/katello-pull-provider-migrate.spec

```diff
@@ -0,0 +1,43 @@
+Name:           katello-pull-provider-migrate
+Version:        1.0.0
+Release:        1%{?dist}
+Summary:        An RPM that migrates katello-agent users to the new pull provider
+BuildArch:      noarch
+
+License:        LGPLv2
+URL:            https://github.com/theforeman/foreman-packaging/packages/client/katello-pull-provider-migrate
+Source0:        %{name}.sh
+
+Requires:       bash
+Requires:       yggdrasil
+Requires:       foreman_ygg_worker
+
+%description
+An RPM that migrates katello-agent users to the new pull provider.
+
+Relies on the existing configuration in rhsm.conf to know which Katello server it should communicate with.
+The RPM install will fail if client is not registered to a Katello server.
+This RPM obtains the relevant values from the existing rhsm.conf, writes the configuration to yggdrasil's config.toml,
+and starts the pull provider agent.
+
+%prep
+
+%build
+
+%install
+rm -rf %{buildroot}
+mkdir -p  %{buildroot}%{_bindir}
+cp %{SOURCE0} %{buildroot}%{_bindir}
+
+%clean
+rm -rf %{buildroot}
+
+%post
+SYSCONFDIR=%{_sysconfdir} %{_bindir}/%{name}.sh
+
+%files
+%{_bindir}/%{name}.sh
```

Comment:
while this is (currently) implemented in shell, I don't see a compelling reason to call the file in `/usr/bin` `.sh`, as we might decide to reimplement it in Python or Go tomorrow.

Also, I think this should be in `sbin`, given the commands need to run as root.

Or maybe, not in any *user* visible dir at all, or are users supposed to run this script on their own too?
If they aren't, `/usr/libexec` is a good place for things that are only executed by tools.

Reply:
It may be nice to leave it user-accessible in the case where the user has messed up their config and wants to reset it to the correct one.

Reply:
Removed the `.sh` and put it in `/usr/sbin`. :+1: